python/cudf/cudf/core/frame.py

```diff
@@ -1246,18 +1246,26 @@ def fillna(
                 for key, value in value.items()
             }
 
-        copy_data = self._data.copy(deep=True)
-
-        for name in copy_data.keys():
+        filled_data = {}
```

Comment:
I can't comment since it's not part of this diff, but I noticed that on line 1241 that can be an else instead of an elif. Normally I'd avoid expanding scope, but for such a tiny change I'd make an exception :) Or just leave a TODO.

Reply:
Done